x/epochs/keeper/epoch.go

```diff
@@ -0,0 +1,84 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"cosmossdk.io/x/epochs/types"
+)
+
+// AddEpochInfo adds a new epoch info. Will return an error if the epoch fails validation,
+// or re-uses an existing identifier.
+// This method also sets the start time if left unset, and sets the epoch start height.
+func (k Keeper) AddEpochInfo(ctx context.Context, epoch types.EpochInfo) error {
+	err := epoch.Validate()
+	if err != nil {
+		return err
+	}
+	// Check if identifier already exists
+	isExist, err := k.EpochInfo.Has(ctx, epoch.Identifier)
+	if err != nil {
+		return err
+	}
+	if isExist {
+		return fmt.Errorf("epoch with identifier %s already exists", epoch.Identifier)
+	}
+
+	// Initialize empty and default epoch values
+	if epoch.StartTime.Equal(time.Time{}) {
+		epoch.StartTime = k.environment.HeaderService.GetHeaderInfo(ctx).Time
+	}
+	epoch.CurrentEpochStartHeight = k.environment.HeaderService.GetHeaderInfo(ctx).Height
+
+	err = k.EpochInfo.Set(ctx, epoch.Identifier, epoch)
+	return err
```

Comment:
why we not return directly k.EpochInfo.Set(ctx, epoch.Identifier, epoch) ?

Reply:
nice catch ty